aiidalab/app.py

```diff
@@ -904,17 +908,23 @@ def _is_compatible(self, app_version: str) -> bool:
             return False  # compatibility indetermined for given version
 
     def _refresh_versions(self) -> None:
+        from packaging.version import parse
+
         self.installed_version = (
             self._get_installed_version()
         )  # only update at this refresh method
+
         self.include_prereleases = self.include_prereleases or (
             isinstance(self.installed_version, str)
+            and is_valid_version(self.installed_version)
```

Comment:
Users can still install apps with invalid version specifiers if they explicitly request it, e.g.

```console
aiidalab install "aiidalab-widgets-base@git+https://github.com/aiidalab/aiidalab-widgets-base@stable"
```
so we need to guard against that here.